data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/PluginBeanFactoryProvider.java

```diff
@@ -61,4 +63,18 @@ public BeanFactory get() {
         final GenericApplicationContext isolatedPluginApplicationContext = new GenericApplicationContext(sharedPluginApplicationContext);
         return isolatedPluginApplicationContext.getBeanFactory();
     }
+
+    public BeanFactory initializePluginSpecificIsolatedContext(Class[] markersToScanForDI) {
+        AnnotationConfigApplicationContext pluginDIContext = new AnnotationConfigApplicationContext();
```

Comment:
We create another BeanFactory above and use that elsewhere. We should only have one isolated context per plugin. But, with this change, we now have two. They both have the same parent.

```
 isolatedPluginApplicationContext.getBeanFactory();
```

Let's consolidate these two.

Reply:
consolidated them. The `get` method is now calling `initializePluginSpecificIsolatedContext` method with an empty class array.